### ISSUE TITLE
Update Ocelot's Configuration without Restarting

### ISSUE THREAD

**Issue by heymega** [ThreeMammals/Ocelot#546]
I noticed this was mentioned back in 2017 for #36 and it seemed Consul was recommended. Hoever, I'm after a simpler solution whereby we can simply change the `ocelot.json` file.

Can we use something like [IOptions](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.options.ioptions-1?view=aspnetcore-2.1) and [IOptionsSnapshot](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.options.ioptionssnapshot-1?view=aspnetcore-2.1) to achieve the refreshing?

I've created this issue as a means of checking whether or not Ocelot currently supports this.
If it does not, then maybe we could discuss whether or not it would be benefitical.

**Comment by TomPallister** [comment]
@heymega yes this is taken care of, yes reloading the configuration will not effect the current request.
